pretrain.py

```diff
@@ -0,0 +1,184 @@
+from transformers import AutoTokenizer
+from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer
+from transformers import DataCollatorForSeq2Seq
+from datasets import Dataset
+import random
+from datasets import load_dataset, concatenate_datasets
+
+import argparse
+import datetime
+import os
+import re
+from typing import Dict
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--batch_size", default=2, type=int, help="Batch size.")
+parser.add_argument("--pretrain_type", default=None, type=str, help="Pretraining method")
+parser.add_argument("--weight_decay", default=0.01, type=float, help="Weight Decay.")
+parser.add_argument("--steps", default=200000, type=int, help="Number of steps.")
+parser.add_argument("--lr", default=2e-5, type=float, help="Learning Rate.")
+parser.add_argument("--model", default="t5-small", type=str, help="Seq to Seq Model")
+parser.add_argument("--seed", default=42, type=int, help="Random seed.")
+parser.add_argument("--threads", default=1, type=int, help="Maximum number of threads to use.")
+parser.add_argument("--ckpt_path", default=None, type=str, help="Checkpoint Path to resume from")
+
+
+def main(args: argparse.Namespace) -> None:
+    # Fix random seeds and threads
+    args.logdir = os.path.join("logs", "{}-{}-{}".format(
+        os.path.basename(globals().get("__file__", "notebook")),
+        datetime.datetime.now().strftime("%Y-%m-%d_%H%M%S"),
+        ",".join(("{}={}".format(re.sub("(.)[^_]*_?", r"\1", k), v) for k, v in sorted(vars(args).items())))
+    ))
+    
+    
+
+
+    if args.pretrain_type == "bt5":
+
+        # BT5: Training T5 on en and ru Wiki dump. 
+        # load parquet files for both languages
+
+        dataset_en = load_dataset("parquet", data_files="data/en.parquet")
+        dataset_ru = load_dataset("parquet", data_files="data/ru.parquet")
+        dataset_en = dataset_en["train"].remove_columns([col for col in dataset_en["train"].column_names if col != "input"])
+        dataset_ru = dataset_ru["train"].remove_columns([col for col in dataset_ru["train"].column_names if col != "input"])
+        assert dataset_en.features.type == dataset_ru.features.type
+        
+        dataset = concatenate_datasets([dataset_en, dataset_ru])
+        dataset = dataset.add_column("label", dataset["input"])
+
+        def get_training_corpus():
+            for start_idx in range(0, len(dataset), 1000):
+                samples = dataset[start_idx : start_idx + 1000]
+                yield samples["input"]
+
+
+    elif args.pretrain_type == "bt5_wmt" or args.pretrain_type == "mt5_wmt":
+
+        # BT5_WMT/MT5_WMT: Training T5/MT5 further on WMT dataset. 
+        # load and process tsv file for WMT News dataset
+
+        dataset = open("./data/news-commentary-v14.en-ru.tsv").read().split("\n")
+        en = []
+        ru = []
+        for line in dataset:
+            en.append(line.split("\t")[0])
+            ru.append(line.split("\t")[1])
```

Comment:
why not:
```suggestion
        en = [line.split("\t")[0] for line in dataset]
        ru = [line.split("\t")[1] for line in dataset]
```
But again, you should create a function for getting n-th column from a dataset.